### PR TITLE
Fix screen artifacts in certain setups

### PR DESCRIPTION
**Pull request by thejch**
#### Describe your PR, what does it fix/add?
An attempt at fixing #3952 and #3946 

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
No

#### Is it ready for merging, or does it need work?
Tested it but could possibly use some some discussion. Also not sure about the implications on performance.

### CODE REVIEW

File: src/render/OpenGL.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ bool CHyprOpenGLImpl::passRequiresIntrospection(CMonitor* pMonitor) {
         return false;
 
     for (auto& w : g_pCompositor->m_vWindows) {
-        if (!w->m_bIsMapped || w->isHidden() || (!w->m_bIsFloating && !g_pCompositor->isWorkspaceSpecial(w->m_iWorkspaceID)))
+        if (!w->m_bIsMapped || w->isHidden())
```
Comment (vaxerski) [inline src/render/OpenGL.cpp:142]: it should be `|| (!floating && new_optimizations)`

**Comment by vaxerski** [comment]
it is my oopsie, yeah.

**Comment by salimp2009** [comment]
```

########################################################################################
AUTOGENERATED HYPR CONFIG.
PLEASE USE THE CONFIG PROVIDED IN THE GIT REPO /examples/hypr.conf AND EDIT IT,
OR EDIT THIS ONE ACCORDING TO THE WIKI INSTRUCTIONS.
########################################################################################

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#


# See https://wiki.hyprland.org/Configuring/Monitors/
#monitor=,preferred,auto,auto
monitor=DP-3,1920x1080@144,0X0,1
# bitdepth,10

# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox

# used for screen sharing
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP

exec-once=/usr/lib/polkit-kde-authentication-agent-1
exec-once=waybar
exec-once=swaybg  -i ~/Pictures/wallpapers/wallhaven-02.png  # -o "DP-1"
exec-once=dunst
exec-once=brave
exec-once=gsettings set org.gnome.desktop.interface color-scheme 'prefer-dark'

#Environment variable; used to be defined in wrappedhl
env = _JAVA_AWT_VM_NONREPARENTING,1
env = XCURSOR_SIZE,24
env = QT_QPA_PLATFORM,wayland
# env = QT_QPA_PLATFORM,wayland;xcb
env = QT_QPA_PLATFORMTHEME,qt5ct
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1

env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = WLR_NO_HARDWARE_CURSORS,1
env = WLR_DRM_DEVICES,/dev/dri/card0
# env = __GL_GSYNC_ALLOWED, 1
# env = __GL_VRR_ALLOWED, 1

# XDG
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_DESKTOP,Hyprland

# Theme
# env = XCURSOR_THEME,Breeze
# env = GTK_THEME,Orchis-Dark-Compact

# Toolkit Backend
# env = SDL_VIDEODRIVER,wayland
# env = GDK_BACKEND,wayland,x11


# this is needed if you specifiy the app to start on a specific workspace
# and needs cleanup.sh so that if you want open another instance on another workspace
# cleanup reset windowrule
# write a cleanup.sh script then activate this and below line
# windowrule=workspace 1 silent,brave
# exec-once=cleanup.sh

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = tr 
    # repeat_rate = 50
    # repeat_delay = 240
    kb_variant =
    kb_model =
    kb_options = caps:escape
    kb_rules =

    follow_mouse = 1

    touchpad {
        natural_scroll = yes 
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = true
    workspace_swipe_min_speed_to_force=4
    workspace_swipe_create_new = false
    workspace_swipe_forever = true
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    layout=master
    sensitivity = 1.8  # for mouse cursor

    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)
#   col.active_border=0xff5e81ac
#   col.inactive_border=0x66333333
    apply_sens_to_raw=0 # whether to apply the sensitivity to raw input (e.g. used by games where you aim using your mouse)
    
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

   rounding = 10
   # blur = yes
   # blur_size = 3
   # blur_passes = 1
   # blur_new_optimizations = on

   blur {
      enabled = 1
      size = 3
      passes = 1
      new_optimizations = true
    }

  #  drop_shadow = yes
  #  shadow_range = 4
  #  shadow_render_power = 3
  #  col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes
    animation = windows, 1, 7, default
    animation = fade, 1, 10, default
    animation = workspaces, 1, 6, default

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more
    # bezier = myBezier, 0.05, 0.9, 0.1, 1.05

   # animation = windowsOut, 1, 7, default, popin 80%
   # animation = border, 1, 10, default
    }

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = 0 # master switch for pseudotiling. 
    force_split=2
    # preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    # new_is_master = true
    new_on_top=true
    no_gaps_when_only = true
}

misc {
    disable_hyprland_logo=true
    disable_splash_rendering=true
    mouse_move_enables_dpms=true
}


# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#executing for more
#device:epic mouse V1 {
#    sensitivity = -0.5
#}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrule=tile,kitty
windowrule=opacity 0.80,alacritty
windowrule=opacity 0.80,rofi
blurls=rofi

windowrule=opacity 0.80,alacritty
windowrule=opacity 0.80,rofi
windowrule=float,title:Calculator
windowrule=center,^(org.gnome.Calculator)$
windowrule=size 500 400,^(org.gnome.Calculator)$

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, RETURN, exec, kitty
bind = $mainMod, b, exec, brave
bind = $mainMod, Q, killactive, 
# bind = SUPERSHIFT,Q,exit,
bind = SUPER,F,fullscreen,1
bind = SUPERSHIFT,F,fullscreen,0
bind = $mainMod, M, exit, 
bind = $mainMod, E, exec, pcmanfm
bind = $mainMod, V, togglefloating, 
bind = $mainMod, SPACE, exec, rofi -show drun
bind = $mainMod, D, exec, dunstctl close 

bind = $mainMod, P, exec, pavucontrol    # somehow Hyprland says this is not installed but it is !!
#bind = $mainMod, J, togglesplit, # dwindle
bind = SUPER,Y,exec,grim -g "$(slurp)" - | swappy -f -


bind=,XF86AudioMute,exec,pactl set-sink-mute @DEFAULT_SINK@ toggle
bind=,XF86AudioLowerVolume,exec,pactl set-sink-volume @DEFAULT_SINK@ -5%
bind=,XF86AudioRaiseVolume,exec,pactl set-sink-volume @DEFAULT_SINK@ +5%
bind=,XF86AudioMicMute,exec,pactl set-source-mute @DEFAULT_SOURCE@ toggle
# bind=,F10,exec,pactl set-source-mute @DEFAULT_SOURCE@ toggle
bind=,F10,exec,pactl set-sink-mute @DEFAULT_SINK@ toggle
bind=,F11,exec,pactl set-sink-volume @DEFAULT_SINK@ -5%
bind=,F12,exec,pactl set-sink-volume @DEFAULT_SINK@ +5%
bind=SUPERSHIFT,E,exec,wlogout
bind=SUPERSHIFT,R,exec,wf-recorder -g "$(slurp)"

bind=SUPERSHIFT,RETURN,layoutmsg,swapwithmaster
bind=SUPER,j,layoutmsg,cyclenext
bind=SUPER,k,layoutmsg,cycleprev
bind=SUPER,h,movefocus,l
bind=SUPER,l,movefocus,r

bind=SUPERSHIFT,left,resizeactive,-40 0
bind=SUPERSHIFT,right,resizeactive,40 0

bind=SUPERSHIFT,h,movewindow,l
bind=SUPERSHIFT,l,movewindow,r
bind=SUPERSHIFT,k,movewindow,u
bind=SUPERSHIFT,j,movewindow,d

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l
bind = $mainMod, right, movefocus, r
bind = $mainMod, up, movefocus, u
bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspacesilent, 1
bind = $mainMod SHIFT, 2, movetoworkspacesilent, 2
bind = $mainMod SHIFT, 3, movetoworkspacesilent, 3
bind = $mainMod SHIFT, 4, movetoworkspacesilent, 4
bind = $mainMod SHIFT, 5, movetoworkspacesilent, 5
bind = $mainMod SHIFT, 6, movetoworkspacesilent, 6
bind = $mainMod SHIFT, 7, movetoworkspacesilent, 7
bind = $mainMod SHIFT, 8, movetoworkspacesilent, 8
bind = $mainMod SHIFT, 9, movetoworkspacesilent, 9
bind = $mainMod SHIFT, 0, movetoworkspacesilent, 10

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:273, movewindow
bindm = $mainMod, mouse:272, resizewindow
```

**Comment by vaxerski** [comment]
fixed myself with 15b282ee0c8b48724b7d068aa95bb1133fdda2d7

**Pull request closed without merging**